toothpick-runtime/src/main/java/toothpick/Toothpick.java

```diff
@@ -121,7 +122,9 @@ public static void closeScope(Object name) {
    */
 
   public static void reset() {
-    MAP_KEY_TO_SCOPE.clear();
+    for (Object name : new ArrayList<>(MAP_KEY_TO_SCOPE.keySet())) {
+      closeScope(name);
+    }
```

Comment:
It works in a similar way to `clear()`: reflecting the state of the hash table at some point.
So we clear the table with the elements available at the moment of starting the operation.

The reason to create a copy is because the iterator is meant to be used only from one thread.